qa/integration-tests/src/test/java/io/zeebe/broker/it/startup/MultiPartitionRecoveryTest.java

```diff
@@ -0,0 +1,131 @@
+          .startEvent("start")
+          .serviceTask("task", t -> t.zeebeJobType("foo"))
+          .endEvent("end")
+          .done();
+
+  @Parameter(0)
+  public Consumer<MultiPartitionRecoveryTest> reprocessingTrigger;
+
+  @Parameter(1)
+  public String name;
+
+  public final ClusteringRule clusteringRule = new ClusteringRule();
+  public final GrpcClientRule clientRule = new GrpcClientRule(clusteringRule);
+  @Rule public RuleChain ruleChain = RuleChain.outerRule(clusteringRule).around(clientRule);
+  @Rule public ExpectedException exception = ExpectedException.none();
+  @Rule public Timeout timeout = new Timeout(120, TimeUnit.SECONDS);
+
+  @Parameters(name = "{index}: {1}")
+  public static Object[][] reprocessingTriggers() {
+    return new Object[][] {
+      new Object[] {
+        (Consumer<MultiPartitionRecoveryTest>) (test -> test.restartBroker(0)), "restart one"
+      },
+      new Object[] {
+        (Consumer<MultiPartitionRecoveryTest>) (test -> test.restartBroker(1)), "restart two"
+      },
+      new Object[] {
+        (Consumer<MultiPartitionRecoveryTest>) (test -> test.restartBroker(2)), "restart three"
+      }
+    };
+  }
+
+  @Test
+  public void shouldDistributeDeploymentAfterRestart() {
+    // given
+    final DeploymentEvent deploymentEvent =
+        clientRule
+            .getClient()
+            .newDeployCommand()
+            .addWorkflowModel(WORKFLOW, "workflow.bpmn")
+            .send()
+            .join();
+
+    // when
+    reprocessingTrigger.accept(this);
+
+    // then
+    clientRule.waitUntilDeploymentIsDone(deploymentEvent.getKey());
+  }
+
+  @Test
+  public void shouldCreateWorkflowInstanceOnAllPartitionsAfterRestart() {
+    // given
+    deploy(WORKFLOW, "workflow.bpmn");
+
+    // when
+    reprocessingTrigger.accept(this);
+
+    // then
+    final var partitionIds = clusteringRule.getPartitionIds();
+
+    Awaitility.await("createInstancesOnAllPartitions")
+        .until(
+            () -> {
+              final var workflowInstanceEvent =
+                  clientRule
+                      .getClient()
+                      .newCreateInstanceCommand()
+                      .bpmnProcessId(PROCESS_ID)
+                      .latestVersion()
+                      .send()
+                      .join();
+
+              final var workflowInstanceKey = workflowInstanceEvent.getWorkflowInstanceKey();
+              final var partitionId = Protocol.decodePartitionId(workflowInstanceKey);
+              partitionIds.remove(Integer.valueOf(partitionId));
+              return partitionIds.isEmpty();
+            });
+  }
+
+  protected void restartBroker(final int nodeId) {
+    clusteringRule.restartBroker(nodeId);
+  }
+
+  private void deploy(final BpmnModelInstance workflowTwoTasks, final String s) {
+    final DeploymentEvent deploymentEvent =
+        clientRule
+            .getClient()
+            .newDeployCommand()
+            .addWorkflowModel(workflowTwoTasks, s)
```

Comment:
Correct the parameter name.

```suggestion
  private void deploy(final BpmnModelInstance workflow, final String s) {
    final DeploymentEvent deploymentEvent =
        clientRule
            .getClient()
            .newDeployCommand()
            .addWorkflowModel(workflow, s)
```